flytekit/clis/sdk_in_container/backfill.py

```diff
@@ -0,0 +1,176 @@
+import typing
+from datetime import datetime, timedelta
+
+import click
+
+from flytekit.clis.sdk_in_container.helpers import get_and_save_remote_with_click_context
+from flytekit.clis.sdk_in_container.run import DateTimeType, DurationParamType
+
+_backfill_help = """
+The backfill command generates and registers a new workflow based on the input launchplan to run an
+automated backfill. The workflow can be managed using the Flyte UI and can be canceled, relaunched, and recovered.
+"""
+
+
+def resolve_backfill_window(
+    from_date: datetime = None,
+    to_date: datetime = None,
+    backfill_window: timedelta = None,
+) -> typing.Tuple[datetime, datetime]:
+    """
+    Resolves the from_date -> to_date
+    """
+    if from_date and to_date and backfill_window:
+        raise click.BadParameter("Cannot use from-date, to-date and backfill_window. Use any two")
+    if not (from_date or to_date):
+        raise click.BadParameter(
+            "One of following pairs are required -> (from-date, to-date) | (from-date, backfill_window) |"
+            " (to-date, backfill_window)"
+        )
+    if from_date and to_date:
+        pass
+    elif not backfill_window:
+        raise click.BadParameter("One of start-date and end-date are needed with duration")
+    elif from_date:
+        to_date = from_date + backfill_window
+    else:
+        from_date = to_date - backfill_window
+    return from_date, to_date
+
+
+@click.command("backfill", help=_backfill_help)
+@click.option(
+    "-p",
+    "--project",
+    required=False,
+    type=str,
+    default="flytesnacks",
+    help="Project to register and run this workflow in",
+)
+@click.option(
+    "-d",
+    "--domain",
+    required=False,
+    type=str,
+    default="development",
+    help="Domain to register and run this workflow in",
+)
+@click.option(
+    "-v",
+    "--version",
+    required=False,
+    type=str,
+    default=None,
+    help="Version for the registered workflow. If not specified it is auto-derived using the start and end date",
+)
+@click.option(
+    "-n",
+    "--execution-name",
+    required=False,
+    type=str,
+    default=None,
+    help="Create a named execution for the backfill. This can prevent launching multiple executions.",
+)
+@click.option(
+    "--dry-run",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Just generate the workflow - do not register or execute",
+)
+@click.option(
+    "--parallel/--serial",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="All backfill can be run in parallel - with max-parallelism",
+)
+@click.option(
+    "--no-execute",
```

Comment:
Can we stop having these double-negatives? [`non-fast`](https://github.com/flyteorg/flytekit/blob/1513773eed118fc3ddacffddf71ee3d30cd0c5cf/flytekit/clis/sdk_in_container/register.py#L105) and [`disable-deck`](https://github.com/flyteorg/flytekit/blob/1513773eed118fc3ddacffddf71ee3d30cd0c5cf/flytekit/core/task.py#L94) are enough proof that we ought to keep things positive.

Reply:
this is case when we should not execute. default is execute i can make it like `--parallel / --serial` -> `--execute / --do-not-execute`

Reply:
I really like this feature in click and I'm of the opinion that we should go with `--execute/--do-not-execute`.

Reply:
done